### ISSUE TITLE
Set up caching of common queries

### ISSUE THREAD

**Issue by eepMoody** [open5e/open5e-api#281]
https://www.django-rest-framework.org/api-guide/caching/ seems to provide an out-of-the-box solution that would serve us nicely, but it would be nice if we could just set it and forget it somewhere central.

https://github.com/Suor/django-cacheops seems to support about the right versions for us, and claims to also just generally make Django faster, might be worth a look.

**Implementation thoughts**
- Due to the nature of our server, caches never need to expire and everything is safe to cache.
- A job to generate some subset of common caches could be run at startup, but this is probably unnecessary.
- In particular, the list views would be highly valuable to cache. Individual items are accessed far more rarely, and respond very quickly.

**Comment by samuelrey** [comment]
Has a decision been made here about CDN vs server-side caching?

**Comment by Sturlen** [comment]
For this situation I'd say a CDN is superior since it avoids using any server resources. It will have to be set at the DNS level though.

**Comment by augustjohnson** [comment]
Yeah strongly recommend just using a CDN, and we currently have cloudflare in-line.  I believe this should be our default solution until a problem arises.

**Comment by Sturlen** [comment]
From what I can see, Cloudflare caching is not currently enabled on the API.
Here's a comparison between fetching all monsters from api.open5e.com and my own mirror. The mirror has cf caching enabled.
![image](https://github.com/open5e/open5e-api/assets/43607012/41bbc68c-8b4d-4604-b86d-9ad0391b0764)

As such I don't think this should be marked as completed just yet.

**Comment by augustjohnson** [comment]
Ok, yes, that is pretty damning. 8 seconds!?!?!? damn. I'll see what I can see.

**Comment by augustjohnson** [comment]
I poked around in cloudflare and couldn't see anything obv.  @Sturlen I may want to sync up with you to troubleshoot this.

**Comment by augustjohnson** [comment]
This may not be as big of a deal after we properly prefetch some of the items, resulting in much less load on the backend.
